### DISCUSSION TITLE
Is there a way to change the ssh-key after the cluster has been deployed?

### DISCUSSION THREAD

**Discussion by gtmobiservices** [kube-hetzner/terraform-hcloud-kube-hetzner#807] [Q&A]
I came across a scenario where I needed to change my ssh key using ssh-keygen and since the kube.tf was pointed to ~/.ssh/id_rsa and ~/.ssh/id_rsa.pub, so terraform plan command failed. Is there a way to change the ssh-key after the cluster has been deployed?

**Comment by StefanIGit** [comment]
Hi, I'm looks for the same option. 
As far as I can tell, it is only possible to for new nodes.
Since it is part of the cloud-init which is only executed on first boot.

So this should be possible by (not tested yet) ;)
- adding the new/additional keys to kube.tf, 
- adding new node pools, 
- apply and wait unti the new nodes are working
- drain and remove the old nodes (in k8s)
- set the old node pools to 0 (not deleting) and apply
